specification/cognitiveservices/data-plane/Language/preview/2022-02-01-preview/textanalytics.json

```diff
@@ -0,0 +1,3027 @@
+        }
+      ],
+      "x-ms-discriminator-value": "EntityLinking"
+    },
+    "AnalyzeTextEntityRecognitionInput": {
+      "properties": {
+        "analysisInput": {
+          "$ref": "#/definitions/MultiLanguageAnalysisInput"
+        },
+        "parameters": {
+          "$ref": "#/definitions/EntitiesTaskParameters"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/AnalyzeTextTask"
+        }
+      ],
+      "x-ms-discriminator-value": "EntityRecognition"
+    },
+    "AnalyzeTextKeyPhraseExtractionInput": {
+      "properties": {
+        "analysisInput": {
+          "$ref": "#/definitions/MultiLanguageAnalysisInput"
+        },
+        "parameters": {
+          "$ref": "#/definitions/PreBuiltTaskParameters"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/AnalyzeTextTask"
+        }
+      ],
+      "x-ms-discriminator-value": "KeyPhraseExtraction"
+    },
+    "AnalyzeTextPiiEntitiesRecognitionInput": {
+      "properties": {
+        "analysisInput": {
+          "$ref": "#/definitions/MultiLanguageAnalysisInput"
+        },
+        "parameters": {
+          "$ref": "#/definitions/PiiTaskParameters"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/AnalyzeTextTask"
+        }
+      ],
+      "x-ms-discriminator-value": "PiiEntityRecognition"
+    },
+    "AnalyzeTextLanguageDetectionInput": {
+      "properties": {
+        "analysisInput": {
+          "$ref": "#/definitions/LanguageDetectionAnalysisInput"
+        },
+        "parameters": {
+          "$ref": "#/definitions/LanguageDetectionTaskParameters"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/AnalyzeTextTask"
+        }
+      ],
+      "x-ms-discriminator-value": "LanguageDetection"
+    },
+    "AnalyzeTextSentimentAnalysisInput": {
+      "properties": {
+        "analysisInput": {
+          "$ref": "#/definitions/MultiLanguageAnalysisInput"
+        },
+        "parameters": {
+          "$ref": "#/definitions/SentimentAnalysisTaskParameters"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/AnalyzeTextTask"
+        }
+      ],
+      "x-ms-discriminator-value": "SentimentAnalysis"
+    },
+    "AnalyzeTextJobsInput": {
+      "properties": {
+        "displayName": {
+          "description": "Optional display name for the analysis job.",
+          "type": "string"
+        },
+        "analysisInput": {
+          "$ref": "#/definitions/MultiLanguageAnalysisInput"
+        },
+        "tasks": {
+          "description": "The set of tasks to execute on the input documents.",
+          "type": "array",
+          "items": {
+            "allOf": [
+              {
+                "$ref": "#/definitions/TaskIdentifier"
+              },
+              {
+                "$ref": "#/definitions/AnalyzeTextLROTask"
+              }
+            ]
+          }
```

Comment:
The generation is a bit cleaner if we move TaskIdentifier to become a property of AnalyzeTextLROTask, and remove the allOf here:

```suggestion
          "items": {
            "$ref": "#/definitions/AnalyzeTextLROTask"
          }
```

Right now we're generating extra type that wraps these properties, so it looks a little unclear that the tasks parameter is polymorphic.

Reply:
Yeah, makes sense.  I kept them separate originally when I was trying to use the same discriminator for too many things, but I've just updated this to add it to the LRO task definition to make it more clear.

Reply:
Is it possible to remove the allOf in this definition and just define as...
```
"items": {
  "$ref": "#/definitions/AnalyzeTextLROTask"
}
```
With the allOf, it creates this extra intermediate type `AnalyzeTextJobInputTasksItem` instead of just setting it to `AnalyzeTextLROTask`.

![image](https://user-images.githubusercontent.com/31998003/149592431-02cba89f-f69a-434b-a18c-575b6e1daedb.png)

![image](https://user-images.githubusercontent.com/31998003/149593349-86b5ec1e-de76-4501-82c8-ea7708c3246c.png)

Reply:
Updated!  Let me know if this helps.

Reply:
Yes, thank you!